File: cpp/palindromy.cpp
#include <iostream>
using namespace std;

int zlicz_znaki(char t[]){
    int i=0;
    while (t[i]!= '\0'){
        i++;
    }
    return i; //zwrócenie rozmiaru tekstu
}

bool czy_palindrom(char wyraz[]){
    int r=zlicz_znaki(wyraz); //długosc wyrazu
    bool czyp = true; //bez tego z return false/true
    
    for(int i=0; i< r/2 ; i++){
        if(wyraz[i] != wyraz[r-1-i]){
          czyp=false;
          break;
          //return false;
        }
    }
    
    return czyp;
    //return true
}

int main(int argc, char **argv)
{
	int r=10;
	char wyraz[r];
	cout<< "Podaj wyraz: ";
	cin.getline(wyraz,r);
	
	if(czy_palindrom(wyraz)) cout<<"To palindrom!";
	else cout <<"To nie jest palindrom!";
	cout<<endl;
	
	return 0;
}

File: python/palindromy.py
def czy_palindrom(wyraz):
    rozmiar=len(wyraz)/2
    for i in range(0, int(rozmiar)):
        if wyraz[i] != wyraz[int(rozmiar)-1-i]:
            return False
    return True

def main(args):
    wyraz = input("Podaj słowo: ")
    
    if(czy_palindrom(wyraz)):
        print("To jest palindrom!")
    else: 
        print("To nie jest palindrom!")

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
